src/V3Param.cpp

```diff
@@ -227,6 +359,76 @@ class ParamVisitor : public AstNVisitor {
             }
         }
     }
+    void relinkPinsByName(AstPin* startpinp, AstNodeModule* modp) {
+        std::map<string, AstVar*> nameToPin;
+        for (AstNode* stmtp = modp->stmtsp(); stmtp; stmtp = stmtp->nextp()) {
+            if (AstVar* varp = VN_CAST(stmtp, Var)) {
+                if (varp->isIO() || varp->isGParam() || varp->isIfaceRef()) {
+                    nameToPin.insert(make_pair(varp->name(), varp));
+                }
+            }
+        }
+        for (AstPin* pinp = startpinp; pinp; pinp = VN_CAST(pinp->nextp(), Pin)) {
+            if (AstVar* varp = pinp->modVarp()) {
+                std::map<string, AstVar*>::const_iterator varIt = nameToPin.find(varp->name());
+                UASSERT_OBJ(varIt != nameToPin.end(), varp,
+                            "Not found in " << modp->prettyNameQ());
+                pinp->modVarp(varIt->second);
+            }
+        }
+    }
+    // Check if parameter setting during instantiation is simple enough for hierarchical verilation
+    void checkSupportedParam(AstNodeModule* modp, AstPin* pinp) const {
+        // InitArray and AstParamTypeDType are not supported because that can not be set via -G
+        // option.
+        if (pinp->modVarp()) {
+            bool supported = false;
+            if (AstConst* constp = VN_CAST(pinp->exprp(), Const)) {
+                supported = !constp->isOpaque();
+            }
+            if (!supported) {
+                pinp->v3error(AstNode::prettyNameQ(modp->origName())
+                              << " has hier_block metacomment, hierarchical verilation"
+                              << " supports only integer/floating point/string parameters.");
```

Comment:
Generally errors don't have trailing periods.

Reply:
Got it!
Done in d76b73beff5f2c6560aaa536dcf6ced1bd4fbbcc